### PR TITLE
Remove Guava

### PR DESCRIPTION
**Pull request by dmikurube**
Like we did move Jackson from embulk-core's top-level class loader in #1437, we'd also like to remove Guava so that embulk-core's Guava does not interfere plugin's Guava.

But unlike Jackson, all Guava usage in Embulk can be replaced with trivial use of Java 8 features. We decided to remove all Guava usage, then.

### CODE REVIEW

File: embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java
Comment (legiangthanh) [inline embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java:41]: I think we have an issue if v has an even integer part, e.g 2.5.

Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java:41]: What's the issue?  If `v` is exact `2.5`,

* `roundedDouble` would be `2.0` from the definition of `Math.rint()`.
* `diff` would be `0.5` from `2.5 - 2.0`.
* `lv` would be `3` from `2.5 + 0.5`.

Then it's the expected `HALF_UP` behavior.

Comment (legiangthanh) [reply embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java:41]: Sorry, I just have a mistake. Actually, the issue happens when v has an integer part is 1.5

`v` is exact `1.5`
rounded double would be `2.0`
`diff` would be `-0.5` from `1.5 - 2.0`
lv would be `1` from `1.5 -0 .5`.
I think we expect 2 in here.

Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java:41]: Ah, probably you're right. I'll have another look into Guava, and have a comparison.

Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java]: Thanks for your good catch. I should have used the sign of `v`, not the sign of `diff`, for `0.5`. Fixed it in: e65f6ecd21a728df893bee57dc47fc691f13470e

**Review by legiangthanh — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Added the core team to review...

**Review by hiroyuki-sato — APPROVED** [review]
LGTM 👍

**Comment by dmikurube** [comment]
We confirmed that the previous v0.10.38 worked well with almost all the plugins we have. It's time to merge this after the 1.5yr long interval, and to release v0.10.39!

**Comment by dmikurube** [comment]
Oops, the test failed after merging the old pull request... The merge might have conflicted semantically with recent changes after this pull request got ready. Let me have a look. https://github.com/embulk/embulk/actions/runs/3627519586/jobs/6117488258

**Pull request merged**
